doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,58 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- private - these APIs are intended for use within boundary of a software component such as
+            driver or subsystem. Private APIs defined in Zephyr tree are not subject to
+            :ref:`api_lifecycle`. Therefore, can be changed or removed at any time.
+- internal - these APIs are intended for use only between certain software components. The context 
+            where the API is called or implemented is well defined. For example, functions prefixed
+            with `arch_` are intended for use by the Zephyr kernel to invoke
+            architecture specific code. Internal APIs may be used in-tree and out-of-tree. Internal
+            APIs defined in Zephyr tree are not subject to :ref:`api_lifecycle`. Therefore, can be changed or removed at any time.
```

Comment:
The statement "Therefore, can be changed or removed at any time." is the exact same as for private, but I think this one needs more boundaries, maybe? -- yes it may be changed or removed at any time, but it should still be documented/tracked somewhere (release notes / migration plan?) as there is a much higher chance that such a change would break "internal" users.

Reply:
Added a note about migration guide.